--- a/demos/qtdemo/menumanager.cpp
+++ b/demos/qtdemo/menumanager.cpp
@@ -286,7 +286,7 @@ void MenuManager::showDocInAssistant(con
     if (this->assistantProcess.state() != QProcess::Running){
         QString app = QLibraryInfo::location(QLibraryInfo::BinariesPath) + QDir::separator();
 #if !defined(Q_OS_MAC)
-        app += QLatin1String("assistant");
+        app += QLatin1String("assistant-qt4");
 #else
         app += QLatin1String("Assistant.app/Contents/MacOS/Assistant");
 #endif
--- a/tools/designer/src/designer/assistantclient.cpp
+++ b/tools/designer/src/designer/assistantclient.cpp
@@ -114,7 +114,7 @@ QString AssistantClient::binary()
 {
     QString app = QLibraryInfo::location(QLibraryInfo::BinariesPath) + QDir::separator();
 #if !defined(Q_OS_MAC)
-    app += QLatin1String("assistant");
+    app += QLatin1String("assistant-qt4");
 #else
     app += QLatin1String("Assistant.app/Contents/MacOS/Assistant");    
 #endif
