diff --git a/tools/assistant/compat/translations/translations.pro b/tools/assistant/compat/translations/translations.pro
index 19fcc06..3daaa79 100644
--- a/tools/assistant/compat/translations/translations.pro
+++ b/tools/assistant/compat/translations/translations.pro
@@ -31,4 +31,4 @@ HEADERS += ../helpwindow.h \
            ../fontsettingsdialog.h
 
 
-TRANSLATIONS=$$[QT_INSTALL_TRANSLATIONS]/assistant_de.ts $$[QT_INSTALL_TRANSLATIONS]/assistant_untranslated.ts
+TRANSLATIONS=$$QT_SOURCE_TREE/translations/assistant_de.ts $$QT_SOURCE_TREE/translations/assistant_untranslated.ts
diff --git a/tools/assistant/translations/qt_help.pro b/tools/assistant/translations/qt_help.pro
index 9f4d7d8..d65029c 100644
--- a/tools/assistant/translations/qt_help.pro
+++ b/tools/assistant/translations/qt_help.pro
@@ -39,12 +39,12 @@ HEADERS += ../lib/qhelpcollectionhandler_p.h \
            ../lib/qhelp_global.h
 
 
-TRANSLATIONS=$$[QT_INSTALL_TRANSLATIONS]/qt_help_de.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qt_help_ja.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qt_help_pl.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qt_help_ru.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qt_help_untranslated.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qt_help_zh_CN.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qt_help_zh_TW.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qt_help_da.ts
+TRANSLATIONS=../../../translations/qt_help_de.ts \
+             ../../../translations/qt_help_ja.ts \
+             ../../../translations/qt_help_pl.ts \
+             ../../../translations/qt_help_ru.ts \
+             ../../../translations/qt_help_untranslated.ts \
+             ../../../translations/qt_help_zh_CN.ts \
+             ../../../translations/qt_help_zh_TW.ts \
+             ../../../translations/qt_help_da.ts
 error("This is a dummy profile to be used for translations ONLY.")
diff --git a/tools/assistant/translations/translations.pro b/tools/assistant/translations/translations.pro
index 4b836e6..35522db 100644
--- a/tools/assistant/translations/translations.pro
+++ b/tools/assistant/translations/translations.pro
@@ -40,11 +40,11 @@ HEADERS += ../tools/assistant/aboutdialog.h \
            ../tools/assistant/searchwidget.h \
            ../tools/assistant/topicchooser.h \
 
-TRANSLATIONS=$$[QT_INSTALL_TRANSLATIONS]/assistant_de.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_ja.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_pl.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_ru.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_untranslated.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_zh_CN.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_zh_TW.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_da.ts
+TRANSLATIONS=../../../translations/assistant_de.ts \
+             ../../../translations/assistant_ja.ts \
+             ../../../translations/assistant_pl.ts \
+             ../../../translations/assistant_ru.ts \
+             ../../../translations/assistant_untranslated.ts \
+             ../../../translations/assistant_zh_CN.ts \
+             ../../../translations/assistant_zh_TW.ts \
+             ../../../translations/assistant_da.ts
diff --git a/tools/assistant/translations/translations_adp.pro b/tools/assistant/translations/translations_adp.pro
index c6f3e81..2f5c893 100644
--- a/tools/assistant/translations/translations_adp.pro
+++ b/tools/assistant/translations/translations_adp.pro
@@ -31,11 +31,11 @@ HEADERS += ../compat/helpwindow.h \
            ../compat/fontsettingsdialog.h
 
 
-TRANSLATIONS=$$[QT_INSTALL_TRANSLATIONS]/assistant_adp_de.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_adp_ja.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_adp_pl.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_adp_ru.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_adp_untranslated.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_adp_zh_CN.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/assistant_adp_zh_TW.ts
+TRANSLATIONS=../../../translations/assistant_adp_de.ts \
+             ../../../translations/assistant_adp_ja.ts \
+             ../../../translations/assistant_adp_pl.ts \
+             ../../../translations/assistant_adp_ru.ts \
+             ../../../translations/assistant_adp_untranslated.ts \
+             ../../../translations/assistant_adp_zh_CN.ts \
+             ../../../translations/assistant_adp_zh_TW.ts
 error("This is a dummy profile to be used for translations ONLY.")
diff --git a/tools/designer/translations/translations.pro b/tools/designer/translations/translations.pro
index c8cd76d..8e42af6 100644
--- a/tools/designer/translations/translations.pro
+++ b/tools/designer/translations/translations.pro
@@ -130,10 +130,10 @@ HEADERS += ../../shared/findwidget/abstractfindwidget.h \
            ../../shared/findwidget/itemviewfindwidget.h \
            ../../shared/findwidget/texteditfindwidget.h
 
-TRANSLATIONS=$$[QT_INSTALL_TRANSLATIONS]/designer_de.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/designer_ja.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/designer_pl.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/designer_ru.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/designer_zh_CN.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/designer_zh_TW.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/designer_untranslated.ts
+TRANSLATIONS=../../../translations/designer_de.ts \
+             ../../../translations/designer_ja.ts \
+             ../../../translations/designer_pl.ts \
+             ../../../translations/designer_ru.ts \
+             ../../../translations/designer_zh_CN.ts \
+             ../../../translations/designer_zh_TW.ts \
+             ../../../translations/designer_untranslated.ts
diff --git a/tools/linguist/linguist/linguist.pro b/tools/linguist/linguist/linguist.pro
index 9106ebe..7fd28b5 100644
--- a/tools/linguist/linguist/linguist.pro
+++ b/tools/linguist/linguist/linguist.pro
@@ -96,11 +96,11 @@ FORMS += statistics.ui \
     finddialog.ui
 RESOURCES += linguist.qrc
 
-TRANSLATIONS=$$[QT_INSTALL_TRANSLATIONS]/linguist_ja.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/linguist_pl.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/linguist_ru.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/linguist_untranslated.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/linguist_zh_CN.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/linguist_zh_TW.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/linguist_de.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/linguist_fr.ts
+TRANSLATIONS=../../../translations/linguist_ja.ts \
+             ../../../translations/linguist_pl.ts \
+             ../../../translations/linguist_ru.ts \
+             ../../../translations/linguist_untranslated.ts \
+             ../../../translations/linguist_zh_CN.ts \
+             ../../../translations/linguist_zh_TW.ts \
+             ../../../translations/linguist_de.ts \
+             ../../../translations/linguist_fr.ts
diff --git a/tools/qtconfig/translations/translations.pro b/tools/qtconfig/translations/translations.pro
index 1f9f572..eaf61d7 100644
--- a/tools/qtconfig/translations/translations.pro
+++ b/tools/qtconfig/translations/translations.pro
@@ -7,8 +7,8 @@ HEADERS        += ../colorbutton.h ../previewframe.h ../previewwidget.h ../mainw
 
 FORMS        = ../mainwindowbase.ui ../paletteeditoradvancedbase.ui ../previewwidgetbase.ui
 
-TRANSLATIONS=$$[QT_INSTALL_TRANSLATIONS]/qtconfig_pl.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qtconfig_ru.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qtconfig_untranslated.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qtconfig_zh_CN.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qtconfig_zh_TW.ts
+TRANSLATIONS=../../../translations/qtconfig_pl.ts \
+             ../../../translations/qtconfig_ru.ts \
+             ../../../translations/qtconfig_untranslated.ts \
+             ../../../translations/qtconfig_zh_CN.ts \
+             ../../../translations/qtconfig_zh_TW.ts
diff --git a/tools/qvfb/translations/translations.pro b/tools/qvfb/translations/translations.pro
index f667bb8..3e27ec8 100644
--- a/tools/qvfb/translations/translations.pro
+++ b/tools/qvfb/translations/translations.pro
@@ -26,8 +26,8 @@ SOURCES =  ../qvfb.cpp \
            ../qwssignalhandler.cpp \
            ../../shared/deviceskin/deviceskin.cpp
 
-TRANSLATIONS=$$[QT_INSTALL_TRANSLATIONS]/qvfb_pl.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qvfb_ru.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qvfb_untranslated.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qvfb_zh_CN.ts \
-             $$[QT_INSTALL_TRANSLATIONS]/qvfb_zh_TW.ts
+TRANSLATIONS=../../../translations/qvfb_pl.ts \
+             ../../../translations/qvfb_ru.ts \
+             ../../../translations/qvfb_untranslated.ts \
+             ../../../translations/qvfb_zh_CN.ts \
+             ../../../translations/qvfb_zh_TW.ts
diff --git a/translations/translations.pri b/translations/translations.pri
index 18f7896..6de6fa9 100644
--- a/translations/translations.pri
+++ b/translations/translations.pri
@@ -26,7 +26,6 @@ QT_TS        = de fr zh_CN untranslated ar es iw ja_JP pl pt ru sk sv uk zh_TW d
 ts-qt.commands = (cd $$QT_SOURCE_TREE/src && $$LUPDATE \
                                     3rdparty/phonon \
                                     3rdparty/webkit \
-                                    activeqt \
                                     corelib \
                                     gui \
                                     network \
@@ -38,10 +37,10 @@ ts-qt.commands = (cd $$QT_SOURCE_TREE/src && $$LUPDATE \
                                     svg \
                                     xml \
                                     xmlpatterns \
-                                -ts $$prependAll($$[QT_INSTALL_TRANSLATIONS]/qt_,$$QT_TS,.ts))
+                                -ts $$prependAll($$QT_SOURCE_TREE/translations/qt_,$$QT_TS,.ts))
 ts-qt.depends = sub-tools
 
-qm-qt.commands = $$LRELEASE $$prependAll($$[QT_INSTALL_TRANSLATIONS]/qt_,$$QT_TS,.ts)
+qm-qt.commands = $$LRELEASE $$prependAll($$QT_SOURCE_TREE/translations/qt_,$$QT_TS,.ts)
 qm-qt.depends = sub-tools
 
 ###### Designer
