--- qt4-x11-4.3.1.orig/src/corelib/arch/qatomic_s390.h
+++ qt4-x11-4.3.1/src/corelib/arch/qatomic_s390.h
@@ -197,10 +197,20 @@
 #endif
 }
 
-#error "fetch-and-add not implemented"
-// int q_atomic_fetch_and_add_int(volatile int *ptr, int value);
-// int q_atomic_fetch_and_add_acquire_int(volatile int *ptr, int value);
-// int q_atomic_fetch_and_add_release_int(volatile int *ptr, int value);
+inline int q_atomic_fetch_and_add_int(volatile int *ptr, int value)
+{
+    return __sync_fetch_and_add(ptr, value);
+}
+
+inline int q_atomic_fetch_and_add_acquire_int(volatile int *ptr, int value)
+{
+    return __sync_fetch_and_add(ptr, value);
+}
+
+inline int q_atomic_fetch_and_add_release_int(volatile int *ptr, int value)
+{
+    return __sync_fetch_and_add(ptr, value);
+}
 
 QT_END_HEADER
 
