author: Brian Nelson <pyro@debian.org>

Ensures qmake generates Makefiles that use the -qt4 tools.

--- a/mkspecs/common/linux.conf
+++ b/mkspecs/common/linux.conf
@@ -23,8 +23,8 @@
 QMAKE_LIBS_OPENGL_QT  = -lGL
 QMAKE_LIBS_THREAD     = -lpthread
 
-QMAKE_MOC             = $$[QT_INSTALL_BINS]/moc
-QMAKE_UIC             = $$[QT_INSTALL_BINS]/uic
+QMAKE_MOC             = $$[QT_INSTALL_BINS]/moc-qt4
+QMAKE_UIC             = $$[QT_INSTALL_BINS]/uic-qt4
 
 QMAKE_AR              = ar cqs
 QMAKE_OBJCOPY         = objcopy
