author: Brian Nelson <pyro@debian.org>

--- a/tools/assistant/lib/qassistantclient.cpp
+++ b/tools/assistant/lib/qassistantclient.cpp
@@ -192,11 +192,11 @@
     : QObject( parent ), host ( QLatin1String("localhost") )
 {
     if ( path.isEmpty() )
-        assistantCommand = QLatin1String("assistant");
+        assistantCommand = QLatin1String("assistant-qt4");
     else {
         QFileInfo fi( path );
         if ( fi.isDir() )
-            assistantCommand = path + QLatin1String("/assistant");
+            assistantCommand = path + QLatin1String("/assistant-qt4");
         else
             assistantCommand = path;
     }
