author: Fathi Boudra <fabo@debian.org>

--- a/tools/designer/src/lib/shared/qdesigner_utils.cpp
+++ b/tools/designer/src/lib/shared/qdesigner_utils.cpp
@@ -488,7 +488,7 @@
         binary += QDir::separator();
         switch (mode) {
         case UIC_GenerateCode:
-            binary += QLatin1String("uic");
+            binary += QLatin1String("uic-qt4");
             break;
         case UIC_ConvertV3:
             binary += QLatin1String("uic3");
