author: Fathi Boudra <fboudra@free.fr>

--- qt4-x11-4.3.1.orig/tools/designer/src/lib/shared/qdesigner_utils.cpp
+++ qt4-x11-4.3.1/tools/designer/src/lib/shared/qdesigner_utils.cpp
@@ -133,7 +133,7 @@
         binary += QDir::separator();
         switch (mode) {
         case UIC_GenerateCode:
-            binary += QLatin1String("uic");
+            binary += QLatin1String("uic-qt4");
             break;
         case UIC_ConvertV3:
             binary += QLatin1String("uic3");
