author: Fathi Boudra <fabo@debian.org>

--- a/src/corelib/kernel/qcoreapplication.cpp
+++ b/src/corelib/kernel/qcoreapplication.cpp
@@ -2061,6 +2061,12 @@ QStringList QCoreApplication::libraryPat
                 app_libpaths->append(installPathPlugins);
         }
 
+        QString kde4PathPlugins = QLibraryInfo::location(QLibraryInfo::LibrariesPath) + QLatin1String("/kde4/plugins");
+        if (QFile::exists(kde4PathPlugins)) {
+            if (!app_libpaths->contains(kde4PathPlugins))
+                app_libpaths->append(kde4PathPlugins);
+        }
+
         // If QCoreApplication is not yet instantiated,
         // make sure we add the application path when we construct the QCoreApplication
         if (self) self->d_func()->appendApplicationPathToLibraryPaths();
