--- a/src/network/qsslsocket_openssl_symbols.cpp
+++ b/src/network/qsslsocket_openssl_symbols.cpp
@@ -247,7 +247,7 @@
         return false;
     }
 #else
-    QLibrary libssl(QLatin1String("ssl"));
+    QLibrary libssl(QLatin1String("ssl.so.0.9.8"));
     if (!libssl.load()) {
         // Cannot find libssl
         qWarning("QSslSocket: cannot find ssl library: %s.",
@@ -255,7 +255,7 @@
         return false;
     }
 
-    QLibrary libcrypto(QLatin1String("crypto"));
+    QLibrary libcrypto(QLatin1String("crypto.so.0.9.8"));
     if (!libcrypto.load()) {
         // Cannot find libcrypto
         qWarning("QSslSocket: cannot find crypto library: %s.",
