--- a/qmake/generators/unix/unixmake2.cpp
+++ b/qmake/generators/unix/unixmake2.cpp
@@ -1440,6 +1440,8 @@
         pkgConfiglibName = "-l" + lname.left(lname.length()-Option::libtool_ext.length());
     }
     t << pkgConfiglibDir << " " << pkgConfiglibName << " ";
+    t << endl;
+    t << "Libs.private: ";
     for(QStringList::ConstIterator it = libs.begin(); it != libs.end(); ++it) {
         t << project->values((*it)).join(" ") << " ";
     }
