Index: b/src/gui/kernel/kernel.pri
===================================================================
--- a/src/gui/kernel/kernel.pri
+++ b/src/gui/kernel/kernel.pri
@@ -112,6 +112,7 @@
             HEADERS += \
                 kernel/qguieventdispatcher_glib_p.h
             QMAKE_CXXFLAGS += $$QT_CFLAGS_GLIB
+            LIBS +=$$QT_LIBS_GLIB
 	}
             SOURCES += \
 		kernel/qeventdispatcher_x11.cpp
@@ -140,6 +141,7 @@
             HEADERS += \
                 kernel/qeventdispatcher_glib_qws_p.h
             QMAKE_CXXFLAGS += $$QT_CFLAGS_GLIB
+	    LIBS +=$$QT_LIBS_GLIB
 	}
 
 
Index: b/src/gui/painting/painting.pri
===================================================================
--- a/src/gui/painting/painting.pri
+++ b/src/gui/painting/painting.pri
@@ -63,6 +63,8 @@
         painting/qpathclipper.cpp \
 	painting/qmemrotate.cpp
 
+LIBS += -lz
+
         DEFINES += QT_RASTER_IMAGEENGINE
         win32:DEFINES += QT_RASTER_PAINTENGINE
         embedded:DEFINES += QT_RASTER_PAINTENGINE
