--- a/src/gui/kernel/kernel.pri
+++ b/src/gui/kernel/kernel.pri
@@ -146,6 +146,7 @@ embedded {
             HEADERS += \
                 kernel/qeventdispatcher_glib_qws_p.h
             QMAKE_CXXFLAGS += $$QT_CFLAGS_GLIB
+             LIBS +=$$QT_LIBS_GLIB
 	}
 
 
--- a/src/gui/painting/painting.pri
+++ b/src/gui/painting/painting.pri
@@ -79,6 +79,8 @@ SOURCES += \
         painting/qtextureglyphcache.cpp \
         painting/qtransform.cpp \
 
+         LIBS += -lz
+
         DEFINES += QT_RASTER_IMAGEENGINE
         win32:DEFINES += QT_RASTER_PAINTENGINE
         embedded:DEFINES += QT_RASTER_PAINTENGINE
