--- qt4-x11-4.4.0~rc1.orig/src/gui/kernel/qapplication_x11.cpp	2008-04-21 19:16:19.000000000 +0300
+++ qt4-x11-4.4.0~rc1/src/gui/kernel/qapplication_x11.cpp	2008-04-27 03:40:39.000000000 +0300
@@ -1743,6 +1743,10 @@
             xfixesFound = X11->ptrXFixesQueryExtension
                           && X11->ptrXFixesQueryVersion
                           && X11->ptrXFixesSetCursorName;
+        } else {
+            X11->ptrXFixesQueryExtension = 0;
+            X11->ptrXFixesQueryVersion = 0;
+            X11->ptrXFixesSetCursorName = 0;
         }
 #else
         bool xfixesFound = true;
@@ -1778,6 +1782,8 @@
         if (xcursorLib.load()) {
             X11->ptrXcursorLibraryLoadCursor =
                 (PtrXcursorLibraryLoadCursor) xcursorLib.resolve("XcursorLibraryLoadCursor");
+        } else {
+            X11->ptrXcursorLibraryLoadCursor = 0;
         }
 #else
         X11->ptrXcursorLibraryLoadCursor = XcursorLibraryLoadCursor;
@@ -1794,6 +1800,10 @@
                 (PtrXineramaIsActive) xineramaLib.resolve("XineramaIsActive");
             X11->ptrXineramaQueryScreens =
                 (PtrXineramaQueryScreens) xineramaLib.resolve("XineramaQueryScreens");
+        } else {
+            X11->ptrXineramaQueryExtension = 0;
+            X11->ptrXineramaIsActive = 0;
+            X11->ptrXineramaQueryScreens = 0;
         }
 #else
         X11->ptrXineramaQueryScreens = XineramaQueryScreens;
