--- a/src/3rdparty/webkit/WebKit.pri
+++ b/src/3rdparty/webkit/WebKit.pri
@@ -26,6 +26,7 @@ CONFIG(release) {
 
 BASE_DIR = $$PWD
 INCLUDEPATH += $$PWD/WebKit/qt/Api
+QMAKE_LFLAGS += -Wl,--version-script,$$BASE_DIR/symbols.filter
 
 #
 # For builds inside Qt we interpret the output rule and the input of each extra compiler manually
--- /dev/null
+++ b/src/3rdparty/webkit/symbols.filter
@@ -0,0 +1,5 @@
+{
+local:
+_ZSt*;
+_ZNSt*;
+};
