--- a/tools/qdoc3/qdoc3.pro
+++ b/tools/qdoc3/qdoc3.pro
@@ -106,3 +106,8 @@ win32 {
         QT_WINCONFIG = debug
     }    
 }
+
+TARGET = qdoc3
+target.path = $$[QT_INSTALL_BINS]
+INSTALLS += target
+
--- a/tools/tools.pro
+++ b/tools/tools.pro
@@ -13,6 +13,7 @@ no-png {
          SUBDIRS += designer
      }
      SUBDIRS     += linguist
+     SUBDIRS     += qdoc3
      wince*: SUBDIRS = qtestlib designer
      unix:!mac:!embedded:contains(QT_CONFIG, qt3support):SUBDIRS += qtconfig
      win32:!wince*:!contains(QT_EDITION, OpenSource|Console):SUBDIRS += activeqt
