author: Thiemo Seufer <ths@debian.org>

Fix compilation for Linux MIPS-I code.

--- a/src/corelib/arch/mips/qatomic32.s
+++ b/src/corelib/arch/mips/qatomic32.s
@@ -7,10 +7,12 @@
 	.ent	q_atomic_test_and_set_int
         .set mips2
 q_atomic_test_and_set_int:
+	.set mips2
 1:	ll   $8,0($4)
 	bne  $8,$5,2f
 	move $2,$6
 	sc   $2,0($4)
+	.set mips0
 	beqz $2,1b
 	nop
 	jr   $31
@@ -24,6 +26,7 @@ q_atomic_test_and_set_int:
 	.ent	q_atomic_test_and_set_acquire_int
         .set mips2
 q_atomic_test_and_set_acquire_int:
+	.set mips2
 1:	ll   $8,0($4)
 	bne  $8,$5,2f
 	move $2,$6
@@ -33,6 +36,7 @@ q_atomic_test_and_set_acquire_int:
 	jr   $31
 	nop
 2:	sync
+	.set mips0
 	jr   $31
 	move $2,$0
         .set mips0
@@ -42,11 +46,13 @@ q_atomic_test_and_set_acquire_int:
 	.ent	q_atomic_test_and_set_release_int
         .set mips2
 q_atomic_test_and_set_release_int:
+	.set mips2
 	sync
 1:	ll   $8,0($4)
 	bne  $8,$5,2f
 	move $2,$6
 	sc   $2,0($4)
+	.set mips0
 	beqz $2,1b
 	nop
 	jr   $31
@@ -60,10 +66,12 @@ q_atomic_test_and_set_release_int:
 	.ent	q_atomic_test_and_set_ptr
         .set mips2
 q_atomic_test_and_set_ptr:
+	.set mips2
 1:	ll   $8,0($4)
 	bne  $8,$5,2f
 	move $2,$6
 	sc   $2,0($4)
+	.set mips0
 	beqz $2,1b
 	nop
 	jr   $31
@@ -77,6 +85,7 @@ q_atomic_test_and_set_ptr:
 	.ent	q_atomic_test_and_set_acquire_ptr
         .set mips2
 q_atomic_test_and_set_acquire_ptr:
+	.set mips2
 1:	ll   $8,0($4)
 	bne  $8,$5,2f
 	move $2,$6
@@ -86,6 +95,7 @@ q_atomic_test_and_set_acquire_ptr:
 	jr   $31
 	nop
 2:	sync
+	.set mips0
 	jr   $31
 	move $2,$0
         .set mips0
@@ -95,11 +105,13 @@ q_atomic_test_and_set_acquire_ptr:
 	.ent	q_atomic_test_and_set_release_ptr
         .set mips2
 q_atomic_test_and_set_release_ptr:
+	.set mips2
 	sync
 1:	ll   $8,0($4)
 	bne  $8,$5,2f
 	move $2,$6
 	sc   $2,0($4)
+	.set mips0
 	beqz $2,1b
 	nop
 	jr   $31
