status: fixed for Qt 4.4.0

--- a/src/corelib/global/qglobal.h
+++ b/src/corelib/global/qglobal.h
@@ -834,13 +834,13 @@
 
 typedef int QNoImplicitBoolCast;
 
-#if defined(QT_ARCH_ARM) || (defined(QT_ARCH_MIPS) && !defined(Q_OS_IRIX))
+#if defined(QT_ARCH_ARM) || (defined(QT_ARCH_MIPS) && (defined(Q_WS_QWS) || defined(Q_OS_WINCE)))
 #define QT_NO_FPU
 #endif
 
 #if defined(QT_COORD_TYPE)
 typedef QT_COORD_TYPE qreal;
-#elif defined(QT_NO_FPU)
+#elif defined(QT_NO_FPU) || defined(QT_ARCH_ARM) || defined(QT_ARCH_WINDOWSCE)
 typedef float qreal;
 #else
 typedef double qreal;
