author: Fathi Boudra <fboudra@free.fr>

Fix arm FTBFS.

--- a/tools/qtestlib/src/qtestcase.h
+++ b/tools/qtestlib/src/qtestcase.h
@@ -208,7 +208,7 @@
     template <typename T1, typename T2>
     bool qCompare(T1 const &, T2 const &, const char *, const char *, const char *, int);
 
-#if defined(QT_COORD_TYPE) || defined(__arm__)
+#if defined(QT_COORD_TYPE) || defined(QT_ARCH_ARM)
     template <>
     inline bool qCompare<qreal, double>(qreal const &t1, double const &t2, const char *actual,
                                  const char *expected, const char *file, int line)
