author: Lennert Buytenhek <buytenh@wantstofly.org>

Add support for EABI ARM platforms to Qt.

--- a/src/corelib/global/qglobal.h
+++ b/src/corelib/global/qglobal.h
@@ -329,7 +329,7 @@
 #  if defined(Q_OS_DARWIN) && __GNUC__ == 3 && (__GNUC_MINOR__ >= 1 && __GNUC_MINOR__ < 3)
 #    define Q_BROKEN_DEBUG_STREAM
 #  endif
-#  if (defined(__arm__) || defined(__ARMEL__)) && !defined(QT_MOC_CPP)
+#  if (defined(__arm__) || defined(__ARMEL__)) && !defined(__ARM_EABI__) && !defined(QT_MOC_CPP)
 #    define Q_PACKED __attribute__ ((__packed__))
 #    define Q_NO_PACKED_REFERENCE
 #  endif
