--- a/src/corelib/global/qlibraryinfo.cpp
+++ b/src/corelib/global/qlibraryinfo.cpp
@@ -67,14 +67,7 @@
 {
 public:
     static QSettings *findConfiguration();
-    static void cleanup()
-    {
-        QLibrarySettings *ls = qt_library_settings();
-        if (ls) {
-            delete static_cast<QSettings *>(ls->settings);
-            ls->settings = 0;
-        }
-    }
+    static void cleanup();
     static QSettings *configuration()
     {
 #ifdef QT_NO_THREAD
@@ -93,6 +86,15 @@
     }
 };
 
+void QLibraryInfoPrivate::cleanup()
+{
+    QLibrarySettings *ls = qt_library_settings();
+    if (ls) {
+        delete static_cast<QSettings *>(ls->settings);
+        ls->settings = 0;
+    }
+}
+
 QLibrarySettings::QLibrarySettings()
 {
     settings = QLibraryInfoPrivate::findConfiguration();
--- a/src/corelib/tools/qhash.h
+++ b/src/corelib/tools/qhash.h
@@ -477,7 +477,7 @@
 }
 
 template <class Key, class T>
-Q_INLINE_TEMPLATE void QHash<Key, T>::duplicateNode(QHashData::Node *node, void *newNode)
+void QHash<Key, T>::duplicateNode(QHashData::Node *node, void *newNode)
 {
     Node *concreteNode = concrete(node);
     if (QTypeInfo<T>::isDummy) {
