--- a/src/sql/drivers/ibase/qsql_ibase.cpp
+++ b/src/sql/drivers/ibase/qsql_ibase.cpp
@@ -467,7 +467,7 @@
 }
 /* char** ? seems like bad influence from oracle ... */
 template<>
-static QList<QVariant> toList<long>(char** buf, int count, long*)
+QList<QVariant> toList<long>(char** buf, int count, long*)
 {
     QList<QVariant> res;
     for (int i = 0; i < count; ++i) {
@@ -622,7 +622,7 @@
 }

 template<>
-static char* fillList<float>(char *buffer, const QList<QVariant> &list, float*)
+char* fillList<float>(char *buffer, const QList<QVariant> &list, float*)
 {
     for (int i = 0; i < list.size(); ++i) {
         double val;
