author: Matthias Klose <doko@debian.org>

--- a/mkspecs/common/g++.conf
+++ b/mkspecs/common/g++.conf
@@ -17,7 +17,7 @@
 QMAKE_CFLAGS_USE_PRECOMPILE += -include ${QMAKE_PCH_OUTPUT_BASE}
 
 QMAKE_CXX		= g++
-QMAKE_CXXFLAGS		+= $$QMAKE_CFLAGS
+QMAKE_CXXFLAGS		+= $$QMAKE_CFLAGS -fpermissive
 QMAKE_CXXFLAGS_DEPS	+= $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON	+= $$QMAKE_CFLAGS_WARN_ON
 QMAKE_CXXFLAGS_WARN_OFF	+= $$QMAKE_CFLAGS_WARN_OFF
